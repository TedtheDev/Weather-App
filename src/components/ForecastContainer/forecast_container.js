import React, { Component } from 'react';
import styled from 'styled-components';
import OpenWeatherAPI from '../../../services/openweathermap';
import SearchBarContainer from '../SearchBarContainer/search_bar_container';
import ForecastTiles from '../ForecastTiles/forecast_tiles';

const ForecastDiv = styled.div`
    grid-area: app;
    display: grid;
    grid-template: 5vh 80vh / auto; 
`;

class Forecast extends Component {
    
    constructor(props) {
        super(props);

        this.state = { forecastData: {}, tilesLoading: false, error: '' };

        this.getForecast = this.getForecast.bind(this);
    }

    async getForecast(cityName) {
        this.setState({ tilesLoading: true })
        const API = OpenWeatherAPI;
        const results = await API.fiveDayAverageCityName(cityName);
        if(results.cod !== '200') {
            this.setState({ tilesLoading: false, error: results.message })
        } else {
            this.setState({ forecastData: results, tilesLoading: false, error: '' });
        }
    }

    render() {
        return (
            <ForecastDiv>
                <SearchBarContainer 
                    getForecast={this.getForecast}
                    error={this.state.error}
                />
                <ForecastTiles 
                    loading={this.state.tilesLoading} 
                    data={this.state.forecastData.list || []}
                />
            </ForecastDiv>
        )
    }
};

export default Forecast;