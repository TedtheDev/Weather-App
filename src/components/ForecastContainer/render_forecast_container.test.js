import React from 'react';
import ForecastContainer from './forecast_container';
import renderer from 'react-test-renderer';

describe('<ForecastContainer />', () => {
    test('renders correctly', () => {
        const component = renderer.create(<ForecastContainer />);

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});