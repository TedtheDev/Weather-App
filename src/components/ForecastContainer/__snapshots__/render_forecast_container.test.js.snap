// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastContainer /> renders correctly 1`] = `
<div
  className="sc-dnqmqq hgyJEZ"
>
  <div
    className="sc-bdVaJa aBlhB"
  >
    <input
      className="sc-bwzfXH hMRMmP"
      onChange={[Function]}
      placeholder="City Name"
      value=""
    />
    <div
      className="sc-bxivhb cfEzOD"
    >
      
    </div>
    <button
      className="sc-htpNat bupPbj"
      onClick={[Function]}
    >
      Get Forecast
    </button>
  </div>
  <div
    className="sc-gzVnrw bwqzJf"
  />
</div>
`;
